src/region.js

```diff
@@ -109,7 +108,8 @@ const Region = MarionetteObject.extend({
   },
 
   _attachView(view, options = {}) {
-    const shouldTriggerAttach = !view._isAttached && isNodeAttached(this.el) && !this._shouldDisableMonitoring();
+    const isNodeAttached = this.Dom.hasEl(document.documentElement, this.el);
```

Comment:
so since the `!view._isAttached` is cheap, I'd like to figure this out so that the `hasEl` is still only called if `view._isAttached` since `hasEl` is more expensive.